File: AppliSportSpringBoot/src/main/java/projet/AppliSport/model/Evenement.java
package projet.AppliSport.model;

import java.time.LocalDate;
import java.util.List;
import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Version;
import javax.validation.constraints.FutureOrPresent;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.PastOrPresent;

import org.springframework.format.annotation.DateTimeFormat;


@Entity
@Table(name="evenement")
@SequenceGenerator(name = "seqEvenement", sequenceName = "seq_evenement", initialValue = 100, allocationSize = 1)
public class Evenement {

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqEvenement")
	@Column(name="id_evenement")
	private Long id;
	
	@NotEmpty
	@Column(name="nom_evenement")
	private String nom;
	
	@OneToMany(mappedBy = "id.utilisateur")
	private List<EvenementUtilisateur> participants;
	
	@PastOrPresent
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	@Column(name="dateDebut_evenement")
	private LocalDate dateDebut;
	
	@FutureOrPresent
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	@Column(name="dateFin_evenement")
	private LocalDate dateFin;
	
	@ManyToOne
	@JoinColumn(name = "evenement_club", foreignKey = @ForeignKey(name = "evenement_club_fk"))
	private Club club;
	
	@Version
	private int version;


	public Evenement() {
		
	}





	public Evenement(Long id, List<EvenementUtilisateur> participants, LocalDate dateDebut, LocalDate dateFin) {
		this.id = id;
		this.participants = participants;
		this.dateDebut = dateDebut;
		this.dateFin = dateFin;
	}


	
	


	public Long getId() {
		return id;
	}





	public void setId(Long id) {
		this.id = id;
	}





	public List<EvenementUtilisateur> getParticipants() {
		return participants;
	}


	public void setParticipants(List<EvenementUtilisateur> participants) {
		this.participants = participants;
	}

	

	public LocalDate getDateDebut() {
		return dateDebut;
	}


	public void setDateDebut(LocalDate dateDebut) {
		this.dateDebut = dateDebut;
	}


	public LocalDate getDateFin() {
		return dateFin;
	}


	public void setDateFin(LocalDate dateFin) {
		this.dateFin = dateFin;
	}


	



	public Club getClub() {
		return club;
	}





	public void setClub(Club club) {
		this.club = club;
	}





	@Override
	public int hashCode() {
		return Objects.hash(id);
	}





	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Evenement other = (Evenement) obj;
		return Objects.equals(id, other.id);
	}

	
	
	
	
	
	

	
	
}

File: SportifyApp/src/app/model/evenement.ts
import { Club } from './club';
import { EvenementUtilisateur } from './evenement-utilisateur';

export class Evenement {
  private _id: number | undefined;
  private _nom: string | undefined;
  private _dateDebut: Date | undefined;
  private _dateFin: Date | undefined;
  private _participants: EvenementUtilisateur[] | undefined;
  private _club: Club | undefined;

  constructor(
    id?: number,
    nom?: string,
    dateDebut?: Date,
    dateFin?: Date,
    participants?: EvenementUtilisateur[],
    club?: Club
  ) {
    this._id = id;
    this._nom = nom;
    this._dateDebut = dateDebut;
    this._dateFin = dateFin;
    this._participants = participants;
    this._club = club;
  }

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter nom
   * @return {string }
   */
  public get nom(): string | undefined {
    return this._nom;
  }

  /**
   * Getter dateDebut
   * @return {Date }
   */
  public get dateDebut(): Date | undefined {
    return this._dateDebut;
  }

  /**
   * Getter dateFin
   * @return {Date }
   */
  public get dateFin(): Date | undefined {
    return this._dateFin;
  }

  /**
   * Getter participants
   * @return {EvenementUtilisateur[] }
   */
  public get participants(): EvenementUtilisateur[] | undefined {
    return this._participants;
  }

  /**
   * Getter club
   * @return {Club }
   */
  public get club(): Club | undefined {
    return this._club;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter nom
   * @param {string } value
   */
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  /**
   * Setter dateDebut
   * @param {Date } value
   */
  public set dateDebut(value: Date | undefined) {
    this._dateDebut = value;
  }

  /**
   * Setter dateFin
   * @param {Date } value
   */
  public set dateFin(value: Date | undefined) {
    this._dateFin = value;
  }

  /**
   * Setter participants
   * @param {EvenementUtilisateur[] } value
   */
  public set participants(value: EvenementUtilisateur[] | undefined) {
    this._participants = value;
  }

  /**
   * Setter club
   * @param {Club } value
   */
  public set club(value: Club | undefined) {
    this._club = value;
  }
}
